### PR TITLE
[Merged by Bors] - feat: When `NNRat.cast` is less/greater than `1`

### PR DESCRIPTION
**Pull request by YaelDillies**
From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Rat/Cast/Order.lean [hunk 1]
```diff
@@ -132,6 +132,10 @@ def castOrderEmbedding : ℚ≥0 ↪o K :=
 @[simp] lemma cast_pos : (0 : K) < q ↔ 0 < q := by norm_cast
 @[norm_cast] lemma cast_lt_zero : (q : K) < 0 ↔ q < 0 := by norm_cast
 @[simp] lemma not_cast_lt_zero : ¬(q : K) < 0 := mod_cast not_lt_zero'
+@[simp, norm_cast] lemma cast_le_one : (p : K) ≤ 1 ↔ p ≤ 1 := by norm_cast
```
Comment (b-mehta) [inline Mathlib/Data/Rat/Cast/Order.lean:135]: Given that `norm_cast` can already prove it, does this need to be tagged as such?

Comment (YaelDillies) [reply Mathlib/Data/Rat/Cast/Order.lean:135]: Maybe not... I guess we don't have `normCastNF`

File: Mathlib/Data/Rat/Cast/Order.lean [hunk 2]
```diff
@@ -132,6 +132,10 @@ def castOrderEmbedding : ℚ≥0 ↪o K :=
 @[simp] lemma cast_pos : (0 : K) < q ↔ 0 < q := by norm_cast
 @[norm_cast] lemma cast_lt_zero : (q : K) < 0 ↔ q < 0 := by norm_cast
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Rat/Cast/Order.lean:133]: Not for this PR, but that's a weird one.

File: Mathlib/Data/Rat/Cast/Order.lean [hunk 3]
```diff
@@ -132,6 +132,10 @@ def castOrderEmbedding : ℚ≥0 ↪o K :=
...
 @[simp] lemma not_cast_lt_zero : ¬(q : K) < 0 := mod_cast not_lt_zero'
+@[simp] lemma cast_le_one : (p : K) ≤ 1 ↔ p ≤ 1 := by norm_cast
+@[simp] lemma one_le_cast : 1 ≤ (p : K) ↔ 1 ≤ p := by norm_cast
+@[simp] lemma cast_lt_one : (p : K) < 1 ↔ p < 1 := by norm_cast
+@[simp] lemma one_lt_cast : 1 < (p : K) ↔ 1 < p := by norm_cast
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Rat/Cast/Order.lean:138]: Do we want similar lemmas for `OfNat.ofNat`?

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

File: Mathlib/Data/Rat/Cast/Order.lean (hunk 2 above)
Comment (YaelDillies) [reply Mathlib/Data/Rat/Cast/Order.lean:133]: I agree [but](https://github.com/leanprover-community/mathlib4/pull/13547#discussion_r1632069488)

**Review by Ruben-VandeVelde — COMMENTED** [review]
maintainer merge

Thanks!

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**